src/EventStore.Projections.Core/Messages/ReaderSubscriptionMessage.cs

```diff
@@ -265,5 +265,15 @@ public class Faulted : SubscriptionMessage {
 				get { return _reason; }
 			}
 		}
+
+		public class ReportProgress : SubscriptionMessage {
+			private static readonly int TypeId = System.Threading.Interlocked.Increment(ref NextMsgId);
+
+			public override int MsgTypeId {
+				get { return TypeId; }
+			}
+
+			public ReportProgress() : base(Guid.NewGuid(), null, null) { }
```

Comment:
IMO we should follow the convention established by the other implementations of `SubscriptionMessage`. `correlationId` and an optional `source` should come from the constructor.

Reply:
Done